builder/oracle/classic/step_security.go

```diff
@@ -15,49 +16,128 @@ type stepSecurity struct{}
 
 func (s *stepSecurity) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
 	ui := state.Get("ui").(packer.Ui)
+	config := state.Get("config").(*Config)
 
-	ui.Say("Configuring security lists and rules to enable SSH access...")
+	commType := ""
+	if config.Comm.Type == "ssh" {
+		commType = "SSH"
+	} else if config.Comm.Type == "winrm" {
+		commType = "WINRM"
+	}
+
+	ui.Say(fmt.Sprintf("Configuring security lists and rules to enable %s access...", commType))
 
-	config := state.Get("config").(*Config)
 	client := state.Get("client").(*compute.ComputeClient)
+	runUUID := uuid.TimeOrderedUUID()
 
-	secListName := fmt.Sprintf("/Compute-%s/%s/Packer_SSH_Allow_%s",
-		config.IdentityDomain, config.Username, config.ImageName)
+	namePrefix := fmt.Sprintf("/Compute-%s/%s/", config.IdentityDomain, config.Username)
+	secListName := fmt.Sprintf("Packer_%s_Allow_%s_%s", commType, config.ImageName, runUUID)
 	secListClient := client.SecurityLists()
 	secListInput := compute.CreateSecurityListInput{
-		Description: "Packer-generated security list to give packer ssh access",
-		Name:        secListName,
+		Description: fmt.Sprintf("Packer-generated security list to give packer %s access", commType),
+		Name:        namePrefix + secListName,
 	}
 	_, err := secListClient.CreateSecurityList(&secListInput)
 	if err != nil {
 		if !strings.Contains(err.Error(), "already exists") {
 			err = fmt.Errorf("Error creating security List to"+
-				" allow Packer to connect to Oracle instance via SSH: %s", err)
+				" allow Packer to connect to Oracle instance via %s: %s", commType, err)
 			ui.Error(err.Error())
 			state.Put("error", err)
 			return multistep.ActionHalt
 		}
 	}
 	// DOCS NOTE: user must have Compute_Operations role
-	// Create security rule that allows Packer to connect via SSH
+	// Create security rule that allows Packer to connect via SSH or winRM
+	var application string
+	if commType == "SSH" {
+		application = "/oracle/public/ssh"
+	} else if commType == "WINRM" {
+		// Check to see whether a winRM security protocol is already defined;
+		// don't need to do this for SSH becasue it is built into the Oracle API.
+		protocolClient := client.SecurityProtocols()
+		winrmProtocol := fmt.Sprintf("WINRM_%s", runUUID)
+		getSecProtocolInput := compute.GetSecurityProtocolInput{
+			Name: namePrefix + winrmProtocol,
+		}
+		protocolInfo, err := protocolClient.GetSecurityProtocol(&getSecProtocolInput)
+		if err == nil {
```

Comment:
what would be the scenario where this already exists?

Reply:
This is leftover from before I forced them to be unique with a UUID, so I guess this scenario won't exist; it's just defensive coding. I can remove it.

Reply:
ahh that makes sense. I think removing it makes sense since it's not really used